orangecontrib/prototypes/widgets/owsplit.py

```diff
@@ -0,0 +1,135 @@
+import Orange.data
+import numpy as np
+from Orange.data import Table
+from Orange.data.util import SharedComputeValue
+from Orange.widgets import gui
+from Orange.widgets.settings import ContextSetting, DomainContextHandler
+from Orange.widgets.widget import OWWidget, Msg, Output, Input
+from AnyQt.QtCore import Qt
+
+from Orange.data import DiscreteVariable, StringVariable
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+
+foo = ["foo,bar", "foo,baz", "foo", "bar,baz", "baz,foo", "baz", "bar"]
+
+
+def values_from_string(var_list, splitter):
+    values = [s.split(splitter) for s in var_list]
+    new_values = sorted(set(val for vals in values for val in vals))
+    mapper = {v: i for i, v in enumerate(new_values)}
+
+    X = np.zeros((len(var_list), len(new_values)))
+    for i, vals in enumerate(values):
```

Comment:
I think this could get slow, but I'm not sure how to optimize it through numpy.

I would think if it could be turned around, so the loop iterates across columns instead of rows. Don't know whether it's possible, though.